extra_scripts/codemirror/addon/hyperlink/hyperlink.js

```diff
@@ -0,0 +1,124 @@
+(function (mod) {
+  if (typeof exports === 'object' && typeof module === 'object') { // Common JS
+    mod(require('../codemirror/lib/codemirror'))
+  } else if (typeof define === 'function' && define.amd) { // AMD
+    define(['../codemirror/lib/codemirror'], mod)
+  } else { // Plain browser env
+    mod(CodeMirror)
+  }
+})(function (CodeMirror) {
+  'use strict'
+
+  const shell = require('electron').shell
+  const yOffset = 2
+
+  class HyperLink {
+    constructor(cm) {
+      this.cm = cm
+      this.lineDiv = cm.display.lineDiv
+
+      this.onMouseDown = this.onMouseDown.bind(this)
+      this.onMouseEnter = this.onMouseEnter.bind(this)
+      this.onMouseLeave = this.onMouseLeave.bind(this)
+      this.onMouseMove = this.onMouseMove.bind(this)
+
+      this.tooltip = document.createElement('div')
+      this.tooltipContent = document.createElement('div')
+      this.tooltipIndicator = document.createElement('div')
+      this.tooltip.setAttribute('class', 'CodeMirror-hover CodeMirror-matchingbracket CodeMirror-selected')
+      this.tooltip.setAttribute('cm-ignore-events', 'true')
+      this.tooltip.appendChild(this.tooltipContent)
+      this.tooltip.appendChild(this.tooltipIndicator)
+      this.tooltipContent.textContent = 'Cmd + click to follow link'
+
+      this.lineDiv.addEventListener('mousedown', this.onMouseDown)
+      this.lineDiv.addEventListener('mouseenter', this.onMouseEnter, {
+        capture: true,
+        passive: true
+      })
+      this.lineDiv.addEventListener('mouseleave', this.onMouseLeave, {
+        capture: true,
+        passive: true
+      })
+      this.lineDiv.addEventListener('mousemove', this.onMouseMove, {
+        passive: true
+      })
+    }
+    getUrl(el) {
+      const className = el.className.split(' ')
+
+      if (className.indexOf('cm-url') !== -1) {
+        const match = /^\((.*)\)|\[(.*)\]|(.*)$/.exec(el.textContent)
+        return match[1] || match[2] || match[3]
+      }
+
+      return null
+    }
+    onMouseDown(e) {
+      const { target, metaKey } = e
+      if (!metaKey) {
```

Comment:
> On Macintosh keyboards, this is the ⌘ Command key. On Windows keyboards, this is the Windows key (⊞ Windows).

I don't know how it looks on MacOS but on Windows and Linux, the meta key is used to open the main menu therefore when you hit meta key + click link, the link does open in the browser but the menu is also displayed which is quite annoying.

I think you will have to change this key to something else. Also you can't set it to <kbd>ctrl</kbd> on windows and linux as it's the key for multi-cursor in codemirror. `Shift perhaps?`